src/ui/public/directives/paginated_selectable_list.js

```diff
@@ -0,0 +1,61 @@
+import _ from 'lodash';
+import uiModules from 'ui/modules';
+import paginatedSelectableListTemplate from 'ui/partials/paginated_selectable_list.html';
+
+const module = uiModules.get('kibana');
+
+module.directive('paginatedSelectableList', function (kbnUrl) {
+
+  return {
+    restrict: 'E',
+    scope: {
+      perPage: '=?',
+      list: '=',
+      listProperty: '=',
+      userMakeUrl: '=?',
+      userOnSelect: '=?'
```

Comment:
Add a check in the controller to see if `userMakeUrl ` && `userOnSelect` have been passed, throw an error if they have. Add a test to check that behavior.